src/Samples/SarifDeferredSample/Program.cs

```diff
@@ -27,7 +27,6 @@ private static void Main(string[] args)
             {
                 // Use 'SarifDeferredContractResolver' and 'JsonPositionedTextReader' to load a geferred version of the same object graph.
                 JsonSerializer serializer = new JsonSerializer();
-                serializer.ContractResolver = (deferred ? SarifDeferredContractResolver.Instance : SarifContractResolver.Instance);
```

Comment:
>SarifDeferredContractResolver [](start = 58, length = 29)

I don't think you can delete this, otherwise the deferred contract resolver won't get invoked. What you need to do, I think, is only set ContractResolver for the deferred case (and not set it to SarifContractResolver in the other case).